pkg/controllermanager/controller/backupentry/backup_entry_reconciler.go

```diff
@@ -0,0 +1,254 @@
+	}
+	// When a BackupEntry deletion timestamp is not set we need to create/reconcile the backup entry.
+	return r.reconcileBackupEntry(be)
+}
+
+func (r *reconciler) reconcileBackupEntry(backupEntry *gardencorev1alpha1.BackupEntry) (reconcile.Result, error) {
+	backupEntryLogger := logger.NewFieldLogger(logger.Logger, "backupentry", fmt.Sprintf("%s", backupEntry.Name))
+	if updateErr := r.updateBackupEntryStatusProcessing(backupEntry, "Reconciliation of Backup Entry state in progress.", 2); updateErr != nil {
+		backupEntryLogger.Errorf("Could not update the status after reconciliation start: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	seedClient, err := getSeedClient(r.ctx, r.client, *backupEntry.Spec.Seed)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	a := newActuator(r.client, seedClient, backupEntry, r.logger)
+	if err := a.Reconcile(r.ctx); err != nil {
+		backupEntryLogger.Errorf("Failed to reconcile backup entry: %+v", err)
+
+		reconcileErr := &gardencorev1alpha1.LastError{
+			Codes:       gardencorev1alpha1helper.ExtractErrorCodes(err),
+			Description: err.Error(),
+		}
+		r.recorder.Eventf(backupEntry, corev1.EventTypeWarning, gardenv1beta1.EventReconcileError, "%s", reconcileErr.Description)
+
+		if updateErr := r.updateBackupEntryStatusError(backupEntry, reconcileErr.Description+" Operation will be retried.", reconcileErr); updateErr != nil {
+			backupEntryLogger.Errorf("Could not update the BackupEntry status after deletion error: %+v", updateErr)
+			return reconcile.Result{}, updateErr
+		}
+		return reconcile.Result{}, errors.New(reconcileErr.Description)
+	}
+
+	if updateErr := r.updateBackupEntryStatusSucceeded(backupEntry, "Backup Entry has been successfully reconciled."); updateErr != nil {
+		backupEntryLogger.Errorf("Could not update the Shoot status after reconciliation success: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	if updateErr := controllerutils.RemoveGardenerOperationAnnotation(r.ctx, retry.DefaultBackoff, r.client, backupEntry); updateErr != nil {
+		backupEntryLogger.Errorf("Could not remove %q annotation: %+v", gardencorev1alpha1.GardenerOperation, updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) deleteBackupEntry(backupEntry *gardencorev1alpha1.BackupEntry) (reconcile.Result, error) {
+	backupEntryLogger := logger.NewFieldLogger(r.logger, "backupentry", fmt.Sprintf("%s", backupEntry.Name))
+	if !sets.NewString(backupEntry.Finalizers...).Has(gardenv1beta1.GardenerName) {
+		backupEntryLogger.Debug("Do not need to do anything as the BackupEntry does not have my finalizer")
+		return reconcile.Result{}, nil
+	}
+
+	gracePeriod := computeGracePeriod(r.deletionGracePeriodHours)
+	present, _ := strconv.ParseBool(backupEntry.ObjectMeta.Annotations[gardencorev1alpha1.BackupEntryForceDeletion])
+	if present || time.Since(backupEntry.DeletionTimestamp.Time) > gracePeriod {
```

Comment:
Does it make sense / is it required to convert the times to UTC?

Reply:
Yes. Nice catch.